### DISCUSSION TITLE
Update Additional Comments and send to requester

### DISCUSSION THREAD

**Discussion by DrMarkX** [Snow-Shell/servicenow-powershell#191] [Q&A] [answered]
Is there a way to use Update-ServiceNowRecord to enter an additional comment that is sent to the requester?
`Update-ServiceNowRecord -Table 'Catalog Task' -ID 'task_number' -Values @{comments = "comments to send to requester"}`

**Comment by gdbarron** [comment]
That looks right to me on the surface. Not working?  Might be to notify the requestor you update the ritm instead.

**Reply by DrMarkX** [reply]
The update command fails with a 403 forbidden error.  I think this is from a ServiceNow update to work notes/additional comments where you must check either _Send To Requestor_ or _Send To Request For_ before a comment can be saved to the Task.

**Reply by DrMarkX** [reply]
I've tested adding a comment to the RITM with the following.
`Update-ServiceNowRecord 'RITM_number' -Values @{comments = "my comments"}`
This does add additional comments to the RITM, but they are not emailed to the requester.  I also noted that the _Send To Requestor_ and _Send To Request For_ checkboxes are in the RITM too.

**Comment by DrMarkX** [comment, marked as answer]
I have figured it out!  You need to use comments, set work_notes to false, notify to true, type to Additional Comments, and add a create_by.

```
if (!$SNOWCred) { $SNOWCred = Get-Credential -Message 'Enter ServiceNow Credentials' }
$snUrl = 'yourcompany.service-now.com'
New-ServiceNowSession -Url $snUrl -Credential $SNOWCred

Update-ServiceNowRecord -Table 'Catalog Task' -ID TASK123456 -Values @{"comments" = "These are sent to the requester.";"notify" = $true;"type" = "Additional Comments";"created_by" = $SNOWCred.UserName}
```
